### DISCUSSION TITLE
gatsby-plugin-typescript option to precompile

### DISCUSSION THREAD

**Discussion by gburgett** [gatsbyjs/gatsby#28938] [Ideas / Feature Requests]
Hello!

I have seen a very significant build speed improvement (50%+) by pre-compiling all my typescript and removing the `gatsby-plugin-typescript` except in development.  My situation is that all content for my site is driven by Contentful, so I almost never change the typescript source code.

I believe this performance boost is due to reduced memory usage.  I am running the build on a Heroku 2X dyno with 1GB of Ram.  I was previously seeing a lot of Swap usage over the course of the build, and am seeing a lot less now.

I'm willing to add to the Gatsby documentation to describe how to precompile your typescript to achieve this savings in production, if you're interested.

**Comment by JustFly1984** [comment]
@gburgett  I’m interested
